catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/BasicTypes.java

```diff
@@ -99,6 +101,8 @@
 
     public static final String VALIDATION_ERRORS = "validation-errors";
 
+    private static final Map<String, AttributeType> ATTRIBUTE_TYPE_MAP = new HashMap<>();
+
```

Comment:
I agree, and we can do your first point, but for your second point, we can't create the `AttributeType`s in a separate method because they are `final`. Alternatively, we can just initialize them at their declaration. I can also move the map creation code to its own method.